ecommerce/models.py

```diff
@@ -646,6 +646,16 @@ def __str__(self):
         """Description for Coupon"""
         return f"Coupon {self.coupon_code} for {self.payment}"
 
+    def validate_unique(self, exclude=None):
+        from ecommerce.filters import CouponUtils
+
+        unique_coupon_code = CouponUtils.is_coupon_code_unique(self.coupon_code)
+        if not unique_coupon_code:
+            raise ValidationError(
+                {"coupon_code": "Coupon code already exists in the plaform."}
+            )
+        super(Coupon, self).validate_unique(exclude=exclude)
```

Comment:
We can use `CouponUtils.validate_unique_coupon_code` to avoid duplication/repetition.